### PR TITLE
Update State Transition Function

### PR DESCRIPTION
**Pull request by terencechain**
This PR updates the [state transition function](https://github.com/ethereum/eth2.0-specs/blob/dcb0244a4f99f50dfa82884a81bd14e1eabb316e/specs/core/0_beacon-chain.md#beacon-chain-state-transition-function) to align with the latest dev

Updated:
- Implemented `ProcessOperations` function 
- Added tests for `ProcessOperations` function 
- Enabled state root verification with a config flag
- Removed operation's upper bound checks in block operation, this should be taken care of in ssz
- Minor name changes to closely align with spec

### CODE REVIEW

File: beacon-chain/blockchain/block_processing_test.go [hunk 1]
```diff
@@ -128,6 +128,7 @@ func TestReceiveBlock_ProcessCorrectly(t *testing.T) {
 		ParentRoot: genesis.ParentRoot,
 		BodyRoot:   bodyRoot[:],
 	}
+	beaconState.Eth1DepositIndex = 0
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/block_processing_test.go]: Why not set this in `setupBeaconChain`?

File: beacon-chain/core/state/transition.go [hunk 2]
```diff
@@ -227,8 +323,11 @@ func ProcessBlock(
 //    process_crosslinks(state)
 //    process_rewards_and_penalties(state)
 //    process_registry_updates(state)
+//    # @process_reveal_deadlines
```
Comment (prestonvanloon) [inline beacon-chain/core/state/transition.go]: ?

Comment (terencechain) [reply beacon-chain/core/state/transition.go]: That's what's written in the python code of the spec, I think it means it's a place holder for a phase 1 function. Want me to remove it?

Comment (prestonvanloon) [reply beacon-chain/core/state/transition.go]: yeah, it's a comment within a comment lol

File: beacon-chain/blockchain/block_processing_test.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/blockchain/block_processing_test.go]: We don't have a good test set up function for it. `GenesisBeaconState` is used for production and it increases `Eth1DepositIndex`. To test correctly this receive block function, we have to reset `Eth1DepositIndex` to 0

Happy to set up test helper for this or we can push it to later

**Pull request merged**